stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -229,7 +229,10 @@ class TaskFutureWaitAsyncContext : public AsyncContext {
 /// only need to do double wide atomics if we need to reach for the
 /// StatusRecord pointers and therefore have to update the flags at the same
 /// time.
-class alignas(sizeof(void*) * 2) ActiveTaskStatus {
+///
+/// On 64 bit systems or if SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION=1, this
+/// field needs to be 16 bytes aligned. Otherwise it needs to be 8 bytes aligned
+class alignas(2 * sizeof(void*)) ActiveTaskStatus {
```

Comment:
It's a little confusing that the comment says we need 16-byte alignment on 32-bit systems with priority escalation, but the `alignas` only provides 8-byte alignment there. Maybe add an extra blurb explaining that we're doing shenanigans in that case. "Because the location of PrivateStorage is not 16-byte aligned on 32-bit, we specify 8-byte alignment and then recover the appropriate alignment with a careful arrangement of the fields" or something like that.